tests/core/test_declarative_error.py

```diff
@@ -0,0 +1,110 @@
+#------------------------------------------------------------------------------
+# Copyright (c) 2022, Nucleic Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+#------------------------------------------------------------------------------
+import sys
+
+from textwrap import dedent
+
+import pytest
+
+from enaml.core.declarative_meta import DeclarativeError
+
+from utils import (
+    compile_source, wait_for_window_displayed, is_qt_available,
+    raises_error_signal
+)
+
+
+pytestmark = pytest.mark.skipif(not is_qt_available(),
+                                reason='Requires a Qt binding')
+
+
+def test_error_during_init(enaml_qtbot):
+    source = dedent("""\
+    from enaml.core.api import Conditional
+    from enaml.widgets.api import Window, Container, Label
+
+    enamldef MyWindow(Window): main:
+        Container:
+            Conditional:
+                condition = None
+                Label:
+                    text = "True"
+
+    """)
+    tester = compile_source(source, 'MyWindow')()
+    with pytest.raises(DeclarativeError) as excinfo:
+        tester.show()
+    assert 'line 4, in MyWindow' in excinfo.exconly()
+    assert 'line 5, in Container' in excinfo.exconly()
+    assert 'line 6, in Conditional' in excinfo.exconly()
+
+
+@pytest.mark.qt_no_exception_capture
```

Comment:
This feels excessive. We are not testing Qt here so simply manually firing the event would be enough.

Reply:
Indeed.